vnext/Desktop/FeatureGate.cpp

```diff
@@ -0,0 +1,31 @@
+#include <FeatureGate.h>
+
+#include <mutex>
+#include <unordered_map>
+
+using std::lock_guard;
+using std::mutex;
+using std::string;
+
+namespace {
+std::unordered_map<string, bool> g_featureGates;
+mutex g_featureGatesMutex;
+} // namespace
+
+namespace Microsoft::React {
+
+void __cdecl SetFeatureGate(string &&name, bool value) noexcept {
+  lock_guard<mutex> guard{g_featureGatesMutex};
+  g_featureGates.insert_or_assign(std::move(name), value);
+}
+
+const bool __cdecl GetFeatureGate(string &&name) noexcept {
+  try {
+    lock_guard<mutex> guard{g_featureGatesMutex};
+    return g_featureGates.at(std::move(name));
+  } catch (const std::out_of_range &) {
```

Comment:
>out_of_range [](start = 22, length = 12)

We should NOT be throwing exceptions for non-found values. Each thrown/caught exception is 100K of instructions or so, perf greatly suffers, and exceptions also complicate debugging. Anyways, use .find or similar Api instead of exception throw/catch.